apps/showcase/src/components/showcase/component-replacement/component-replacement-pres.component.ts

```diff
@@ -22,48 +22,31 @@ const ONE_DAY_IN_MS = 24 * 60 * 60 * 1000;
   encapsulation: ViewEncapsulation.None,
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class ComponentReplacementPresComponent implements OnChanges, DynamicConfigurable<ComponentReplacementPresConfig> {
-
-  /** Configuration stream based on the input and the stored configuration*/
-  public config$: Observable<ComponentReplacementPresConfig>;
-
-  @ConfigObserver()
-  private readonly dynamicConfig$: ConfigurationObserver<ComponentReplacementPresConfig>;
+export class ComponentReplacementPresComponent implements DynamicConfigurableWithSignal<ComponentReplacementPresConfig> {
+  private readonly c11nService = inject(C11nService);
 
   /** Input configuration to override the default configuration of the component*/
-  @Input()
-  public config: Partial<ComponentReplacementPresConfig> | undefined;
+  public config = input<Partial<ComponentReplacementPresConfig>>();
 
-  /** Observable of the presenter that we want to use, processed by the c11n directive */
-  public presenter$!: Observable<Type<DatePickerInputPresContext>>;
-
-  public dateFormControl = new FormControl<string | null>(this.formatDate(Date.now() + 7 * ONE_DAY_IN_MS));
+  @O3rConfig()
+  public readonly configSignal = configSignal(
+      this.config,
+      COMPONENT_REPLACEMENT_PRES_CONFIG_ID,
+      COMPONENT_REPLACEMENT_PRES_DEFAULT_CONFIG
+    );
 
-  constructor(
-    private readonly c11nService: C11nService,
-    @Optional() configurationService: ConfigurationBaseService
-  ) {
-    this.dynamicConfig$ = new ConfigurationObserver<ComponentReplacementPresConfig>(COMPONENT_REPLACEMENT_PRES_CONFIG_ID, COMPONENT_REPLACEMENT_PRES_DEFAULT_CONFIG, configurationService);
-    this.config$ = this.dynamicConfig$.asObservable();
-    this.loadPresenter();
-    this.dynamicConfig$.next({ ...this.config, datePickerCustomKey: 'exampleDatePickerFlavorHebrew' });
-  }
+  public datePickerComponent: Signal<Type<DatePickerInputPresContext> | undefined>;
 
-  private formatDate(dateTime: number) {
-    return formatDate(dateTime, 'yyyy-MM-dd', 'en-GB');
-  }
+  public dateFormControl = new FormControl<string | null>(this.formatDate(Date.now() + 7 * ONE_DAY_IN_MS));
 
-  private loadPresenter() {
-    this.presenter$ = this.config$.pipe(
+  constructor() {
+    this.datePickerComponent = toSignal(toObservable(this.configSignal).pipe(
```

Comment:
can't you do it with a compute() ?

Reply:
As discussed `getPresenter` returns a rxjs operator so I need to convert the configSignal into an observable